src/components/SelectableMainTable.tsx

```diff
@@ -0,0 +1,180 @@
+        <>
+          <CheckboxInput
+            label=""
+            labelClassName="multiselect-checkbox"
+            checked={isAllSelected}
+            indeterminate={isSomeSelected && !isAllSelected}
+            onChange={isSomeSelected ? selectNone : selectPage}
+          />
+          <ContextualMenu
+            position="left"
+            title="Multiselect"
+            toggleAppearance="base"
+            toggleClassName="has-icon u-no-margin--bottom"
+            toggleLabel={<Icon name="chevron-down" />}
+            toggleProps={{
+              "aria-label": "multiselect rows",
+            }}
+            links={[
+              {
+                children: `Select all ${itemName}s on this page`,
+                onClick: selectPage,
+              },
+              {
+                children: `Select all ${parentName} ${itemName}s`,
+                onClick: selectAll,
+              },
+            ]}
+          />
+        </>
+      ),
+      className: "select select-header",
+    },
+    ...(headers ?? []),
+  ];
+
+  const rowsWithCheckbox = rows.map((row) => {
+    const isRowSelected = selectedNames.includes(row.name ?? "");
+    const isRowProcessing = processingNames.includes(row.name ?? "");
+
+    const toggleRow = () => {
+      const newSelection = isRowSelected
+        ? selectedNames.filter((candidate) => candidate !== row.name)
+        : [...selectedNames, row.name ?? ""];
+      setSelectedNames(newSelection);
+    };
+
+    row.columns = [
+      {
+        content: (
+          <CheckboxInput
+            label=""
+            labelClassName="u-no-margin--bottom"
+            checked={isRowSelected}
+            onChange={toggleRow}
+            disabled={isRowProcessing}
+          />
+        ),
+        role: "rowheader",
+        "aria-label": `Select ${row.name ?? ""}`,
+        className: "select",
+      },
+      ...(row.columns ?? []),
+    ];
+
+    row.className = classnames(row.className, {
+      "selected-row": isRowSelected,
+      "processing-row": isRowProcessing,
+    });
+
+    return row;
+  });
+
+  const selectionState = isSomeSelected
+    ? [
+        {
+          className: "select-notification",
+          expanded: true,
+          expandedContent: (
+            <Notification
+              borderless
+              className="u-no-margin--bottom"
+              title="Selection"
+            >
+              {isAllSelected ? (
+                <>
+                  All <b>{allNames.length}</b> {itemName}s of this {parentName}{" "}
+                  are selected.{" "}
+                  <Button
+                    appearance="link"
+                    className="u-no-margin--bottom u-no-padding--top"
+                    onClick={selectNone}
+                  >
+                    Clear selection
+                  </Button>
+                </>
+              ) : (
+                <>
+                  <b>{selectedNames.length}</b> {itemName}s selected{" "}
```

Comment:
And here as well, the singular vs plural:

![image](https://user-images.githubusercontent.com/56583786/230141832-b43f812b-61d7-4af8-b0a8-f4fce1dba1d3.png)